inbm/dispatcher-agent/dispatcher/source/constants.py

```diff
@@ -25,16 +26,16 @@ class SourceParameters:
 
 @dataclass(kw_only=True, frozen=True)
 class ApplicationAddSourceParameters:
-    gpg_key_uri: str
-    gpg_key_name: str
     file_name: str
```

Comment:
**Suggestion:** Validate the file_name to ensure it ends with '.list' since it is a convention for APT source list files. [validation]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    @dataclass(kw_only=True, frozen=True)
    class ApplicationAddSourceParameters:
        file_name: str = field(init=False)
        source_list_file_name: str = field(default=None)

        def __post_init__(self):
            if not self.source_list_file_name.endswith('.list'):
                raise ValueError("file_name must end with '.list'")
            self.file_name = self.source_list_file_name
```